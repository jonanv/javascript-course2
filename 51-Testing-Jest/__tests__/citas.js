import Citas from "../js/classes/Citas";

describe('Probar la clase de citas', () => {

    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Juan',
            telefono: '32323',
            fecha: '10-12-2025',
            hora:'10:00',
            sintomas: 'Duermen'
        };

        citas.agregarCita(citaObj);

        // Prueba
        expect(
            citas
        ).toMatchSnapshot();
    });

    test('Actualizar cita', () => {
        const citaObjActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'Juan',
            telefono: '32323',
            fecha: '10-12-2025',
            hora:'10:30',
            sintomas: 'Duermen'
        };

        citas.editarCita(citaObjActualizada);

        // Prueba
        expect(
            citas
        ).toMatchSnapshot();
    });

    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        // Prueba
        expect(
            citas
        ).toMatchSnapshot();
    });
});