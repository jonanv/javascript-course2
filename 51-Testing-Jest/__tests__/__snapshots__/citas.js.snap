// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2025",
      "hora": "10:30",
      "id": 1752435347738,
      "mascota": "Nuevo nombre",
      "propietario": "Juan",
      "sintomas": "Duermen",
      "telefono": "32323",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2025",
      "hora": "10:00",
      "id": 1752435347738,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Duermen",
      "telefono": "32323",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
